source/validations.ts

```diff
@@ -0,0 +1,150 @@
+// /source/validations.ts
+// The validation functions
+
+import type { Request } from 'express'
+
+/**
+ * An error thrown/returned when a validation error occurs.
+ */
+class ValidationError extends Error {
+	name: string
+	code: string
+	message: string
+
+	/**
+	 * The code must be a string, in snake case and all capital, that starts with
+	 * the substring `ERR_ERL_`.
+	 *
+	 * The message must be a string, starting with a lowercase character,
+	 * describing the issue in detail.
+	 */
+	constructor(code: string, message: string) {
+		super(message)
+
+		// `this.constructor.name` is the class name
+		this.name = this.constructor.name
+		this.code = code
+		this.message = message
+	}
+}
+
+/**
+ * Runs the given validation check, and conveys the error message to the user.
+ *
+ * @param shouldRun {boolean} - Whether or not the check should even run.
+ * @param check {Function} - The validation check to run.
+ * @param error {ValidationError} - The error to convey in case the validation check fails.
+ *
+ * @returns {void}
+ */
+const runCheck = (
+	shouldRun: boolean,
+	check: () => boolean,
+	error: ValidationError,
+): void => {
+	if (!shouldRun) return
+
+	// Run the check. If it fails, then return the error message to the user.
+	if (!check()) {
+		const formattedMessage = `express-rate-limit: ${error.code} - ${error.message}`
+		console.error(formattedMessage)
```

Comment:
IMO, this formatting should be done inside the `ValidationError` constructor, e.g. 

```js
super(`express-rate-limit: ${error.code} - ${error.message}`)
```

And then we could either do `console.error(error.message)` here, or `console.error(error)` to include the full stack trace. I'm a bit on the fence about that one.. do you think stack traces will be more annoying or helpful?